modyn/tests/trainer_server/internal/trainer/test_pytorch_trainer.py

```diff
@@ -192,16 +192,26 @@ def get_training_info(
         lr_scheduler_config = {
             "name": "StepLR",
             "source": "PyTorch",
+            "step_every": "batch",
             "optimizers": ["default"] if num_optimizers == 1 else ["opt1"],
             "config": {"step_size": 10},
         }
     elif lr_scheduler == "custom":
         lr_scheduler_config = {
             "name": "custom_scheduler",
             "source": "Custom",
+            "step_every": "batch",
             "optimizers": ["default"] if num_optimizers == 1 else ["opt1", "opt2"],
             "config": {},
         }
+    elif lr_scheduler == "torch_cosine":
+        lr_scheduler_config = {
+            "name": "CosineAnnealingLR",
+            "source": "PyTorch",
+            "step_every": "batch",
+            "optimizers": ["default"] if num_optimizers == 1 else ["opt1"],
+            "config": {"T_max": "MODYN_NUM_BATCHES"},
```

Comment:
I find this way of encoding that configuration option a bit weird. I don't have a better idea though as I don't now what potential other options the config dictionary could hold here. Let's keep that for now :)

Reply:
Yes, I also am not sure if this is the best option, but we somehow need to offer a "runtime" replacement here for a very particular use case, so for now it should be ok...